src/app/modules/core/components/map/map.component.ts

```diff
@@ -19,146 +19,176 @@ import { deepCopy } from '../../../../shared/extensions/object.DeepCopy';
-				})
-			}).on('click', this.swapElement(mySpill['mylist'], 2, 3));
-			//L.DomUtil.removeClass(this.map._container, 'crosshair-cursor-enabled');
-			this.mapReady = false;
-			this.markers.push(marker);
-			//this.checkZoom(this.map.getZoom())//, this.markers)
-			this.StudyAreaService.setSelectedStudyArea(this.markers);
-			//this.MapService.result = mySpill;
-			if (this.methodType === "downstream") {
-				let mySite = mySpill;
-				let e = this.StudyAreaService.spillLocation.getLatLng();
-				this.MarkerClick(mySite['mylist'], e.lat, e.lng, 'downstream', ['streamStatsgage', 'flowline'], 1000);
-				//this.barButtonOptions_downstream.active = true;
-			} else if (this.methodType === "upstream") {
-				let mySite = mySpill;
-				let e = this.StudyAreaService.spillLocation.getLatLng();
-				this.MarkerClick(mySite['mylist'], e.lat, e.lng, 'upstream', ['streamStatsgage'], 1000);
-				//this.barButtonOptions_upstream.active = true;
-			} */
-		} else { }
-	}
-
-	public MarkerClick(e, lat, lng, cond, option, len) {
-		e[0]['value']['coordinates'] = [lng, lat]; // add lat long
-		e[1]['value'] = cond;
-		e[3]['value'] = option;
-		//for upstream only
-		if (e[2].value instanceof Array) { //if array, set limit of 100, copy parameters into a tuple, else do nothing since it should be already set
-		  e[2].value[0].value = len;
-		  e[2].value.splice(1, 1);
-		  let myvar = new parameter(e[2].value[0]);
-		  e[2].value = myvar;
-		}
-	
-		/* this.NavigationService.getRoute("networktrace",e)
-		  .toPromise().then(data => {
-			let myreturn;
-			let polyline;
-			if (cond == 'upstream') {
-			  myreturn = this.MapService.getUpstream(data);
-			  //this.barButtonOptions_upstream.active = false;
-			} else {
-			  myreturn = this.MapService.getDownstream(data);
-			  var myStyle = {
-				"color": "#FF3333",
-				"weight": 3,
-				"opacity": 0.60
-			  }
-			  this.MapService.addPolyLine(data);
-			  polyline = L.geoJSON(this.MapService.streamArray, { style: myStyle });
-			  if (typeof polyline === 'undefined') { } else {
-				this.markers.push(polyline);
-			  }
-			}
-			this.markers.push(myreturn);
-			this.barButtonOptions_downstream.active = false;
-			this.barButtonOptions_downstream.disabled = true;
-			var group = L.featureGroup(this.markers);
-			this.map.fitBounds(group.getBounds());
-			for (var i = 0; i < this._MapService.lastnode.length; i++) {
-			  this.markers.push(this._MapService.lastnode[i])
-			}
-			this.setStep(2);
-		  })
-		  .catch((err) => {
-			console.log("error: ", err.message);
-			this.sendAlert(err, "Navigation Services");
-			this.barButtonOptions_downstream.active = false;
-			this.barButtonOptions_downstream.buttonColor = 'primary';
-			this.markers = [];
-		  });
-		}; */
-	
-	
+  public onMouseClick(evnt: any) {
+    this.MapService.AddLayer(evnt.latlng);
+    this.sm("Layer added to map!!!");
+
+  }
+
+  public addMarker(e) {
+    if (this.MapService.CurrentZoomLevel > 9 && this.mapReady === true) {
```

Comment:
Lets clean this up/out. If we aren't using it/ working on it, lets remove it to clean up the clutter.